### PR TITLE
overlord/ifacestate: register all security backends with the repository

### PR DESCRIPTION
**Pull request by zyga**
The new security backends will require one more thing from the repository. Like interfaces they will
have to  be registered before they can be used . This branch registers all the known backends with
the repository owned by the interface manager. It also adds a way to create the manager with
additional interfaces for upcoming tests.

### CODE REVIEW

**Comment by stolowski** [comment]
Looks sensible. +1

File: overlord/ifacestate/handlers.go [hunk 1]
```diff
@@ -182,7 +182,7 @@ func (m *InterfaceManager) removeProfilesForSnap(task *state.Task, _ *tomb.Tomb,
 	}
 
 	// Remove security artefacts of the snap.
-	if err := removeSnapSecurity(task, snapName); err != nil {
+	if err := removeSnapSecurity(task, snapName, m.repo); err != nil {
```
Comment (pedronis) [inline overlord/ifacestate/handlers.go]: should this be m.removeSnapSecurity instead ?

Comment (zyga) [inline overlord/ifacestate/handlers.go]: Indeed, it feels like it makes more sense this way now.

Comment (zyga) [inline overlord/ifacestate/handlers.go]: Done

**Pull request merged**